### PR TITLE
1867: First pass on merger

### PR DESCRIPTION
**Pull request by jenf**
Mostly working, needs token reduction and share purchasing

### CODE REVIEW

File: lib/engine/step/g_1867/merge.rb [hunk 1]
```diff
@@ -92,14 +101,100 @@ def move_assets(from, to)
...
+            owner = corporation.owner
+            owners << owner
+            @game.close_corporation(corporation)
+
+            share = target.shares.last
+
+            if target.shares.first.president && owner.percent_of(target) == 10
+              # give the 10% back
+              presidency = target.shares.first
+              @game.share_pool.move_share(owner.shares_of(target).first, presidency.owner)
```
Comment (ryandriskel) [inline lib/engine/step/g_1867/merge.rb:147]: Seems there are a number of ways to do this. In 18CO, I used `share.transfer(...)`. When should we use `@game.share_pool` instead?

Comment (tobymao) [reply lib/engine/step/g_1867/merge.rb:147]: not sure, can both accomplish the same thing?

Comment (jenf) [reply lib/engine/step/g_1867/merge.rb:147]: i can rework this later, i'll need to do the later on the case that people are buying shares when they're at 10% anyway

Comment (jenf) [reply lib/engine/step/g_1867/merge.rb:147]: I remember, I added it for 1817 to avoid making that public :)

**Pull request merged**